dali/pipeline/workspace/workspace.h

```diff
@@ -27,6 +27,15 @@
 
 namespace dali {
 
+/**
+ * @brief Used to specify the shape and type of Output
+ * that a Workspace can hold.
+ */
+struct OutputDesc {
+  kernels::TensorListShape<> shape;
+  TypeInfo type;
```

Comment:
```suggestion
  DALIDataType type_id;
```
...should be enough. You can always get a reference to `dali::TypeInfo` using `TypeTable::GetTypeInfo`.
BTW - we should remove usage of TypeInfo - it shouldn't be passed around by value...

Reply:
I think that should be done as a separate PR. For now we can leave TypeInfo as it is used directly with set_type and TypeTable is also a heavy thing to access.